pkg/types/types.go

```diff
@@ -0,0 +1,24 @@
+package types
+
+import (
+	"database/sql"
+	"time"
+)
+
+type ProviderCluster struct {
+	ClusterID   string
+	AccountID   string
+	SatelliteID string
+	MetaData    sql.NullString
+	Spec        sql.NullString
+	Status      sql.NullString
```

Comment:
As mentioned in the previous review, the types should not use package proprietary types. Also we are expecting sub-structures here and not strings

Reply:
Yes am aware of it as it was not clear how it looks like I kept it as String and will change it as we implement it.

Reply:
Just put empty structs